stellargraph/mapper/full_batch_generators.py

```diff
@@ -421,37 +421,22 @@ def __init__(self, G, name=None, sparse=True, transform=None):
         G.check_graph_for_ml()
 
         # extract node, feature, and edge type info from G
-        self.node_list = list(G.nodes())
-
-        self.features = G.node_features(self.node_list)
+        node_types = list(G.node_types)
+        if len(node_types) != 1:
+            raise ValuError(
```

Comment:
```suggestion
            raise ValueError(
```

Reply:
I guess I should have a test for this new branch 😄

Reply:
Done (and reworded the message slightly).